internal/engine/buildcontrol/target_queue.go

```diff
@@ -90,8 +90,24 @@ func NewImageTargetQueue(ctx context.Context, iTargets []model.ImageTarget, stat
 	}, nil
 }
 
-func (q *TargetQueue) Results() store.BuildResultSet {
-	return q.results
+// New results that we built with a BuildHandler. Omits results
+// that were re-used previous builds.
+func (q *TargetQueue) NewResults() store.BuildResultSet {
+	newResults := store.BuildResultSet{}
+	for id, result := range q.results {
+		if q.IsDirty(id) {
```

Comment:
I understand why this works but the verbiage is confusing ("IsDirty" makes me think it STILL needs a build??). Not a big deal to leave as-is but i wonder if there exists a comment that would clarify?
```suggestion
		if q.IsDirty(id) {  // i.e. WASDirty() -- if True, we built this target afresh
```

Reply:
changed it to a private method and renamed it isBuilding